### PR TITLE
initial implementation

### PR DESCRIPTION
**Pull request by the-mikedavis**
as the docs have it, beeline makes a supervision tree

```
Supervisor
├── HealthChecker*
└── StageSupervisor
    ├── Producer*
    └── Consumer
```

for each `Beeline.start_link/2`

the Supervisor is a regular ole' one-for-one restart-permanent supervisor

the StageSupervisor is a bit special:

- it uses the one-for-all strategy which will terminate the producer if the consumer crashes (we want this behavior to enforce ordering)
- it uses the restart-transient restart strategy to ensure that if the GenStages are stuck on an event, it won't take down the entire supervision tree (it won't even take down the health checkers)

### CODE REVIEW

File: test/beeline/dummy_producer_test.exs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  @fixture Beeline.DummyProducerFixture
+
+  setup do
+    [beeline_pid: start_supervised!({@fixture, self()})]
+  end
+
+  test "the dummy handler can handle events" do
+    events = [%{foo: "bar"}, %{foo: "bar"}, %{foo: "bar"}]
+
+    :ok = Beeline.test_events(events, @fixture)
```
Comment (the-mikedavis) [inline test/beeline/dummy_producer_test.exs:55]: :woohoo:! testing utilities

File: test/fixtures/good_handler.ex [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+          connection: Beeline.Fixtures.SpearClient,
+          stream_name: opts[:grpc_stream_name],
+          adapter: :volley
+        ]
+      ],
+      get_stream_position: {__MODULE__, :get_stream_position, []},
+      subscribe_after: 0,
+      context: %__MODULE__{opts: Map.new(opts)}
+    )
+  end
```
Comment (the-mikedavis) [inline test/fixtures/good_handler.ex:51]: a good example for a start_link/2 with multiple producers

File: mix.exs [hunk 3]
```diff
@@ -46,7 +46,9 @@ defmodule Beeline.MixProject do
     [
       {:gen_stage, "~> 1.0"},
       {:nimble_options, "~> 0.1"},
-      {:kelvin, "~> 0.3", optional: true},
+      {:jason, "~> 1.0"},
+      {:health_checker, "~> 2.2", organization: "cuatro"},
```
Comment (the-mikedavis) [inline mix.exs:50]: we can't publish this package publicly until we get rid of this dep

I'm thinking about rolling health_checker into this library and making new appsignal+honeycomb exporter libraries. It'd be a good bit of work but probably worth it to be able to keep everything open-sourceable

I could also be interested in re-naming health_checker. The main reason I see not to just go ahead and publish that publicly now is that it doesn't really deserve to squat on the name health_checker (IMHO) in the public hex.

Comment (tonyvanriet) [reply mix.exs:50]: indeed. it is not as general-purpose as the name suggests. time to tony-fy that name or encode it with some cleverness algorithm.

**Review by tonyvanriet — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -11,11 +11,16 @@ jobs:
...
       MIX_ENV: test
+      EVENTSTORE_TCP_HOST: localhost
+      EVENTSTORE_GRPC_HOST: localhost
 
     steps:
     - name: Checkout
       uses: actions/checkout@v2
 
+    - name: Stand up docker-compose services
+      run: docker-compose up -d
```
[the remark this answers is not in the document]
Comment (the-mikedavis) [reply .github/workflows/ci.yml:22]: the syntax to run stuff through the app container is a bit awkward but it's definitely do-able to have that nice setup as you say

and yep your grokking is :heavy_check_mark:

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -11,11 +11,16 @@ jobs:
     runs-on: ubuntu-16.04
```
[the remark this answers is not in the document]
Comment (the-mikedavis) [reply .github/workflows/ci.yml:11]: ah yeah I just copied this from projection_event_handler

we have to start caring about the v16.04 deprecation soon don't we

**Pull request merged**